synapse/storage/databases/main/client_ips.py

```diff
@@ -351,7 +351,63 @@ def _devices_last_seen_update_txn(txn):
         return updated
 
 
-class ClientIpStore(ClientIpBackgroundUpdateStore):
+class ClientIpWorkerStore(ClientIpBackgroundUpdateStore):
```

Comment:
I was unsure which of these should be the base class? It seems we do have a few layouts:

* worker -> background update -> main
* worker -> main
* (worker, background update) -> main
* background update -> worker -> main

Reply:
I think it depends on if functions in the background updater depend on stuff from the worker or vice versa :shrug: